doc: additionals and improvements on helm migration steps (#3554)
operator-framework/operator-sdk: website/content/en/docs/building-operators/helm/migration.md

---

> title: Helm migration for the new Layout

Reviewer: ```suggestion
title: Migrating Legacy Projects
```

> linkTitle: Migration Guide

Reviewer: ```suggestion
linkTitle: Migrating Legacy Projects
```

### What is new

> To bring a little context, projects are now scaffold using

Reviewer: ```suggestion
Projects are now scaffold using:
```

> - [kustomize][kustomize] to perform the configurations 

Reviewer: ```suggestion
- [kustomize][kustomize] to manage Kubernetes resources needed to deploy your operator
```

> - Makefile with helpers which brings more flexibility and allows customizations

Reviewer: ```suggestion
- A `Makefile` with helpful targets for build, test, and deployment, and to give you flexibility to tailor things to your project's needs.
```

> - Protect by default via [kube-auth-proxy][kube-auth-proxy] 
> - Prometheus metrics configured via [kustomize][kustomize] 

Reviewer: Being a little pedantic here. Prometheus metrics themselves aren't configured via kustomize. However the metrics server, kube-rbac-proxy, and service monitor are:
```suggestion
- Updated metrics configuration using [kube-auth-proxy][kube-auth-proxy], a `--metrics-addr` flag, and [kustomize][kustomize]-based deployment of a Kubernetes `Service` and prometheus operator `ServiceMonitor`
```

 
### Creating a new project

In Kubebuilder-style projects, CRD groups are defined using two different flags

> (`--group` and `--domain).

Author: ```suggestion
(`--group` and `--domain`).
```

 
### Creating a new project

> Let's check our domain first for we create a new project which will have the same `<group>.<domain>/<version>` API's. We can find our domain value in the CRD's, see:
> 
> ```yaml
> ...
>   group: cache.example.com
> ...
> ```
> 
> Then, let's create a new project with the same domain (`example.com`):

Reviewer: ```suggestion
In Kubebuilder-style projects, CRD groups are defined using two different flags
(`--group` and `--domain).

When we initialize a new project, we need to specify the domain that _all_ APIs in
our project will share, so before creating the new project, we need to determine which
domain we're using for the APIs in our existing project.

To determine the domain, look at the `spec.group` field in your CRDs in the
`deploy/crds` directory.

The domain is everything after the first DNS segment. Using `cache.example.com` as an
example, the `--domain` would be `example.com`.

So let's create a new project with the same domain (`example.com`):
```

The below example is correct based on a domain of `example.com`, so that needs to be fixed.

I think we should show the `init` call separate from `create api`. That way the instructions are the same for all project types:

`````
```sh
$ mkdir nginx-operator
$ cd nginx-operator
$ operator-sdk init --plugins=helm --domain=example.com
```

Now that we have our new project initialized, we need to re-create each of our APIs. 
Using our API example from earlier (`cache.example.com`), we'll use `cache` for the
`--group` flag.

For each API in the existing project, run:
```sh
$ operator-sdk create api \
    --group=cache \
    --api-version=<apiVersion> \
    --kind=<Kind> \
    --helm-chart=<path_to_existing_project>/helm-charts/<chart>
```

`````
